### DISCUSSION TITLE
PR creates merge commit but pushes all commits from branch to master

### DISCUSSION THREAD

**Discussion by shellby1** [community/community#16932] [Projects and Issues]
Branch A has 10 commits
PR creates "merge commit" and pushes it to master
but also pushes all commits from branch A into master
thereby PR always pushes branch commits + 1 into master instead one "merge commit"
![image](https://user-images.githubusercontent.com/30357425/169360611-55ce9209-eb1c-47d8-a943-8a3cbd1ad90c.png)
![image](https://user-images.githubusercontent.com/30357425/169360367-2dcfd00f-d380-4475-a39a-80443a387803.png)

git docs says that only one commit should be in master
[About pull request merges](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges)

**Comment by janbrasna** [comment]
That's how merge works. The merge commit is in fact the only one added/created; and with it all of its history arrives;)

More reading:
- https://www.atlassian.com/git/tutorials/using-branches/git-merge
- https://www.atlassian.com/git/tutorials/using-branches/merge-strategy
